content/what-is-jsx.md

```diff
@@ -0,0 +1,255 @@
+component and will pass `MyComponent` as a variable reference into the
+`createElement` function.
+
+This is the point in the documentation where the JSX specification and the
+JSX implementation usually become hopelessly intertwined.
+In the following image we can see where the distinction occurs.
+
+![A file with JSX blocks is transpiled into JavaScript with createElement calls. These createElement calls are then interpreted by the JSX implementation](jsx-transpilation-and-interpretation.svg)
+
+The JSX specification defines how a JavaScript transpiler should translate JSX blocks
+into a series of `createElement` calls. But the `createElement` function
+comes from a JSX _implementation_ and will determine how the actual components
+need to be defined. We will go into the specific details about how the complate
+implementation below, but first we want to look at a few more details about the
+JSX implementation.
+
+<!--
+	here I want HTML so that I can define an anchor element to the section and
+	link to it!
+-->
+
+### Interpolation with JavaScript Expressions
+
+Everything which appears in a `{ }` block will be interpreted as a JavaScript
+expression.
+
+```jsx
+<MyComponent title={title}>
+    <p>{description}</p>
+</MyComponent>
+```
+
+will become
+
+```javascript
+createElement(MyComponent, { title: title },
+	createElement("p", null, description));
+```
+
+This means that we can use JavaScript expressions to express conditionals
+
+```jsx
+<MyComponent title={title || "unknown"}>
+    {description && (
+        <p>{description}</p>
+    )}
+</MyComponent>
+```
+
+or for a loop
+
+```jsx
+<List>
+    {items.map(item => (
+        <span>{item}</span>
+    ))}
+</List>
+```
+
+### Special Cases for Boolean Attributes
+
+Attributes without an explicit value in a JSX expression are interpreted as
+having the boolean value `true`. This means that the following two JSX blocks
+are identical:
+
+```jsx
+<button disabled>
+```
+
+```jsx
+<button disabled={true}>
+```
+
+### Fragment provided by JSX Implementation
+
+JSX always expects exactly one root element in an expression, but in some cases
+we want to return several at once. To do this, a JSX implementation will
+provide a `Fragment` component with provides an artificial hierarchy level.
+
+```jsx
+<dl>
+	{items.map(({ term, description }) => (
+		<Fragment>
+			<dt>{term}</dt>
+			<dd>{description}</dd>
+		</Fragment>
+	))}
+</dl>
+```
+
+### Trying it out
+
+You can try out how JSX translation works in the
+[Babel online compiler](https://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=DwWQngwg9gtgDlAdgU0QFwARoJZoDbIC8ARABLJ55QYDqUATngCbEB8AUAJDBytX3IYGbHADOAVxjAA9L3Yzw0eElRpWQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=7.7.3&externalPlugins=).
```

Comment:
```suggestion
[Babel online compiler](https://babeljs.io/repl#?presets=react&code_lz=DwWQngwg9gtgDlAdgU0QFwARoJZoDbIC8ARABLJ55QYDqUATngCbEB8AUAJDBytX3IYGbHADOAVxjAA9L3Yzw0eElRpWQA).
```

I've removed unused URI parameters, because I'm just that pedantic.